pandas/tests/groupby/test_function.py

```diff
@@ -768,7 +768,15 @@ def test_nsmallest():
 
 @pytest.mark.parametrize(
     "data, groups",
-    [([0, 1, 2, 3], [0, 0, 1, 1]), ([0], [0])],
+    [
+        ([0, 1, 2, 3], [0, 0, 1, 1]),
+        ([0], [0]),
+        *[
```

Comment:
i find this pattern difficult to read.  is there a viable alternative?

Reply:
maybe add a new param `@pytest.mark.parametrize("dtype", [None] + tm.ALL_INT_NUMPY_DTYPES)` and then

```
if dtype is not None:
    data = np.array(data, dtype=dtype)
    groups = np.array(groups, dtype=dtype)
```

Reply:
Yes, I can do that.